core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -264,37 +264,47 @@ protected[core] object WhiskWebActionsApi extends Directives {
   }
 
   /**
-   * Finds the content-type in the header list and maps it to a known media type. If it is not
-   * recognized, construct a failure with appropriate message.
+   * Finds the content-type in the header list and ensures that it is a valid format. If it is not
+   * valid, construct a failure with appropriate message.
+   * If the content-type header is missing, then return the supplied defaultType
    */
   private def findContentTypeInHeader(headers: List[RawHeader],
                                       transid: TransactionId,
                                       defaultType: MediaType): Try[MediaType] = {
     headers.find(_.lowercaseName == `Content-Type`.lowercaseName) match {
       case Some(header) =>
         MediaType.parse(header.value) match {
-          case Right(mediaType: MediaType) =>
-            // lookup the media type specified in the content header to see if it is a recognized type
-            MediaTypes.getForKey(mediaType.mainType -> mediaType.subType).map(Success(_)).getOrElse {
-              // this is a content-type that is not recognized, reject it
-              Failure(RejectRequest(BadRequest, Messages.httpUnknownContentType)(transid))
-            }
-          case _ => Failure(RejectRequest(BadRequest, Messages.httpUnknownContentType)(transid))
+          case Right(mediaType: MediaType) => Success(mediaType)
+          case _                           => Failure(RejectRequest(BadRequest, Messages.httpUnknownContentType)(transid))
         }
       case None => Success(defaultType)
     }
   }
 
+  private def isJsonFamily(mt: MediaType) = {
+    mt == `application/json` || Option(mt.value).map(_.endsWith("+json")).exists(identity)
```

Comment:
Can `mt.value` be `null`? 

I don't think it can. Should it then just be 

```scala
mt == `application/json` || mt.value.endsWith("+json")
```

Reply:
I admit that I didn't check - I carried over the unapply from the previous branch but turned it into an action. It does look like the parser will not do that.